gips/data/modis/modis.py

```diff
@@ -205,47 +203,70 @@ def fetch(cls, asset, tile, date):
             listing = urllib.urlopen(mainurl).readlines()
 
-        success = False
+        available = []
         for item in listing:
             # screen-scrape the content of the page and extract the full name of the needed file
             # (this step is needed because part of the filename, the creation timestamp, is
             # effectively random).
             if cpattern.search(item):
                 if 'xml' in item:
                     continue
-                name = cpattern.findall(item)[0]
-                url = ''.join([mainurl, '/', name])
-                outpath = os.path.join(cls.Repository.path('stage'), name)
-
-                with utils.error_handler("Asset fetch error", continuable=True):
-                    # tinkering:
-                    # chunk size & stream=True in req
-                    # cookies / cache auth
-
-                    if mainurl[0:3] == 'ftp':
-                        connection = urllib2.urlopen(url)
-                        with open(outpath, 'wb') as fd:
-                            fd.write(connection.read())
-
-                    else: # http
-                        kw = {'timeout': 10}
-                        if asset not in cls._skip_auth:
-                            kw['auth'] = (username, password)
-                        response = requests.get(url, **kw)
-
-                        if response.status_code != requests.codes.ok:
-                            print('Download of', name, 'failed:', response.status_code, response.reason,
-                                  '\nFull URL:', url, file=sys.stderr)
-                            return # might as well stop b/c the rest will probably fail too
-
-                        with open(outpath, 'wb') as fd:
-                            for chunk in response.iter_content():
-                                fd.write(chunk)
-                    utils.verbose_out('Retrieved %s' % name, 2)
-                    success = True
-
-        if not success:
-            VerboseOut('Unable to find remote match for %s at %s' % (pattern, mainurl), 4)
+                basename = cpattern.findall(item)[0]
+                url = ''.join([mainurl, '/', basename])
+                available.append({'basename': basename, 'url': url})
+
+        if len(available) == 0:
+            VerboseOut(
+                'Unable to find remote match for {} at {}'
+                .format(pattern, mainurl),
+                4
+            )
+        return available
+
+
+    @classmethod
+    def fetch(cls, asset, tile, date):
+        available_assets = cls.query_service(asset, tile, date)
+        success = False
+        retrieved_filenames = []
+        for asset_info in available_assets:
+            basename = asset_info['basename']
+            url = asset_info['url']
+            outpath = os.path.join(cls.Repository.path('stage'), basename)
+
+            with utils.error_handler("Asset fetch error", continuable=True):
```

Comment:
optional:  in the error message, consider calling out by name which asset had the error